workflows/epitopeprediction.nf

```diff
@@ -184,28 +185,25 @@ workflow EPITOPEPREDICTION {
     ========================================================================================
     */
 
-    // perform the check requested models on the protein and variant files
-    // we have to perform it on all alleles that are given in the sample sheet
-    ch_variants_protein_models = ch_samples_uncompressed
-        .variant
-        .mix(ch_samples_uncompressed.protein)
-        .map { meta_data, file -> meta_data.alleles }
-        .splitCsv(sep: ';')
-        .collect()
-        .toList()
-        .combine(ch_samples_uncompressed.variant.first())
-        .map { it -> tuple(it[0].unique(), it[-1])}
-
+    // perform the check requested models on the variant files
     EPYTOPE_CHECK_REQUESTED_MODELS(
-        ch_variants_protein_models,
+        ch_samples_uncompressed.variant
+            .mix(ch_samples_uncompressed.protein),
+        ch_prediction_tool_versions
+    )
+
+    // perform the check requested models on the protein files
+    EPYTOPE_CHECK_REQUESTED_MODELS_PROTEIN(
+        ch_samples_uncompressed.variant
+            .mix(ch_samples_uncompressed.protein),
         ch_prediction_tool_versions
```

Comment:
Why is this process executed 2x here? Looks like the input is exactly the same. Maybe I missed something